src/wp-includes/functions.php

```diff
@@ -8582,3 +8582,93 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Get supported data types for meta and option values.
+ *
+ * This is a subset of the PHP data types that can be used for
+ * meta and option values. See the documentation for PHP's 'settype()'.
+ *
+ * @since 6.3.0
+ *
+ * @return array List of supported value types.
+ */
+function wp_get_db_value_types() {
+	return array(
+		'boolean',
+		'bool',
+		'integer',
+		'int',
+		'float',
+		'string',
+		'array',
+		'object',
+	);
+}
+
+/**
+ * Set the type of a meta or option value that was retrieved from the database.
+ *
+ * If `$value_type` is empty, the value is returned unchanged. If `$value_type` is not one
+ * of the expected types, a user error is triggered and the value is returned unchanged.
+ *
+ * @since 6.3.0
+ *
+ * @param mixed  $value      Raw value from the database.
+ * @param string $value_type The expected value type.
+ * @return mixed The value with the requested type.
+ */
+function wp_settype_to_value_from_db( $value, $value_type ) {
```

Comment:
This name reads a little off - and `_settype_` is maybe too coupled to implementation. 

What about one of these?
- `wp_set_db_value_to_type()`
- `wp_set_value_from_db_to_type()`
- `wp_convert_db_value()`
- `wp_convert_db_value_to_type()`
- `wp_convert_value_from_db()`
- `wp_set_db_value_type()`